WordPress/src/main/java/org/wordpress/android/ui/accounts/signup/SettingsUsernameChangerFragment.kt

```diff
@@ -0,0 +1,176 @@
+            setView(layout)
+            setPositiveButton(
+                    string.settings_username_changer_confirm_dialog_positive_action
+            ) { _, _ -> saveUsername(username) }
+            setNegativeButton(android.R.string.cancel, null)
+            create()
+        }.show().also { alertDialog ->
+            // The change username button is disabled at start.
+            val changeUsernameButton = alertDialog.getButton(AlertDialog.BUTTON_POSITIVE).apply {
+                isEnabled = false
+            }
+
+            // Toggles the enabled property of the button based on the correctness of the username being typed.
+            usernameControl.addTextChangedListener(object : TextWatcher {
+                override fun afterTextChanged(editable: Editable?) {
+                    editable?.let {
+                        changeUsernameButton.isEnabled = (it.toString() == username)
+                    }
+                }
+
+                override fun beforeTextChanged(s: CharSequence?, start: Int, count: Int, after: Int) {
+                }
+
+                override fun onTextChanged(s: CharSequence?, start: Int, before: Int, count: Int) {
+                }
+            })
+            usernameControl.requestFocus()
+        }
+    }
+
+    private fun saveUsername(username: String) {
+        showProgress()
+        val payload = PushUsernamePayload(
+                username, KEEP_OLD_SITE_AND_ADDRESS
+        )
+        mDispatcher.dispatch(AccountActionBuilder.newPushUsernameAction(payload))
+    }
+
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onUsernameChanged(event: OnUsernameChanged) {
+        if (event.isError) {
+            AnalyticsTracker.track(SIGNUP_EMAIL_EPILOGUE_UPDATE_USERNAME_FAILED)
+            AppLog.e(
+                    T.API, "SettingsUsernameChangerFragment.onUsernameChanged: " +
+                    event.error.type + " - " + event.error.message
+            )
+            endProgress()
+            showErrorDialog(SpannableStringBuilder(getString(string.signup_epilogue_error_generic)))
+        } else if (event.username != null) {
+            val result = Bundle().apply { putString(RESULT_USERNAME, event.username) }
+            dialogController.confirm(result)
+            AnalyticsTracker.track(Stat.SIGNUP_EMAIL_EPILOGUE_UPDATE_USERNAME_SUCCEEDED)
+        }
+    }
+
+    fun showProgress() {
+        if (progressDialog == null || progressDialog?.window == null || progressDialog?.isShowing == false) {
+            progressDialog = ProgressDialog(context).apply {
+                isIndeterminate = true
+                setCancelable(true)
+                setMessage(getString(string.settings_username_changer_progress_dialog))
+            }
+        }
+        try {
+            if (progressDialog?.isShowing == false) {
+                progressDialog?.show()
+            }
+        } catch (e: WindowManager.BadTokenException) {
+            AppLog.e(T.NUX, e)
```

Comment:
I am not sure if this is the right way to use the `AppLog` within this feature. I saw this usage in the signup functionality so I did the same here since the class lives with the same package due to its base class having most of its functionality as protected.

Reply:
;) Great question! 
We have two logging mechanism.
1. AppLog - local only logs. These are used in most cases in which we want to know what is going on when the user reaches support, but isn't critical so we don't need to sent it into Sentry.
2. CrashLoggingUtils - these logs are sent into Sentry. We can sent both fatal and non-fatal logs. However, these logs should be used only where it makes sense to store them into Sentry as there are some quota limits.

So to answer your question, I think using AppLog is the right choice. I'd just consider using T.SETTINGS as it seems like a user experience issue. Wdyt?